Sources/Commands/Utilities/SymbolGraphExtract.swift

```diff
@@ -82,7 +82,22 @@ public struct SymbolGraphExtract {
         if includeSPISymbols {
             commandLine += ["-include-spi-symbols"]
         }
-        
+
+        // If this target or any dependencies is is built with C++ interop
+        // enabled then swift-symbolgraph-extract must also enable C++ interop.
+        var cxxInterop = module.cxxInterop()
+        if !cxxInterop {
+            for module in try module.recursiveTargetDependencies() {
+                if module.cxxInterop() {
+                    cxxInterop = true
+                    break
+                }
+            }
+        }
+        if cxxInterop {
+            commandLine += ["-cxx-interoperability-mode=default"]
```

Comment:
Some projects use a non-default compat version (e.g. `swift-5.9`), we should forward the actual version here.

Reply:
I dont see anything in SwiftPM that currents sets a non-"default" value and based on the PackageModel I dont see how a user can even select a "non-default compat version"

Reply:
It seems like this support was removed in #6583

Reply:
It is still possible to pass a custom C++ interop compat version, for example:
```
            swiftSettings: [
                .unsafeFlags(["-cxx-interoperability-mode=upcoming-swift"])
            ]
```